motioneye/uploadservices.py

```diff
@@ -881,6 +882,79 @@ def _refresh_credentials(self, refresh_token):
         }
 
 
+class Webdav(UploadService):
+    NAME = 'webdav'
+
+    def __init__(self, camera_id):
+        self._server = None
+        self._port = None
+        self._username = None
+        self._password = None
+        self._location = None
+
+        UploadService.__init__(self, camera_id)
+
+    def _request(self, url, method, body=None):
+        base64string = b64encode(f'{self._username}:{self._password}'.encode()).decode('ascii')
+        headers = {'Authorization': f'Basic {base64string}'}
+        if body is not None:
+            headers.update('Content-Length', len(body))
+        self.debug(f'request: {method} {url}')
+        request = urllib.request.Request(url, data=body, headers=headers)
+        request.get_method = lambda: method
+        try:
+            utils.urlopen(request)
+        except urllib.HTTPError as e:
+            if method == 'MKCOL' and e.code == 405:
+                self.debug(
+                    'MKCOL failed with code 405, this is normal if the folder exists'
+                )
+            else:
+                raise e
+
+    def _make_dirs(self, path):
+        dir_url = self._server
+        for folder in path.strip('/').split('/'):
+            dir_url = dir_url + folder + '/'
+            self._request(dir_url, 'MKCOL')
+
+    def test_access(self):
+        try:
+            test_path = self._location.strip('/') + '/' + str(time.time())
```

Comment:
This strips as well the leading slash from the location/path, so that invalid URLs are generated, like `hostname.orgpath/to/dir`. As only the trailing slash is re-added, and the leading slash is explicitly mentioned in the tooltip for the location input, I suggest to use `rstrip` here:
```suggestion
            test_path = self._location.rstrip('/') + '/' + str(time.time())
```